packages/@lightningjs/ui-components/src/utils/index.js

```diff
@@ -743,6 +743,51 @@ export function createConditionalZContext(component, zOffset) {
   }
 }
 
+/**
+ * Runs a side effect function when any values of specified properties on a component change.
+ * @param {Object} options - defines the `component`, `watchProps`, and `sideEffect` options
+ * @param {lng.Component} options.component - Lightning component on which property changes will watched
+ * @param {String[]} options.watchProps - properties that when their value changes a side effect function is invoked
+ * @param {Function} options.sideEffect - function to be invoked when a watched property's value changes
+ * @returns {lng.Component}
+ */
+export function watchForUpdates({
+  component = lng.Component,
```

Comment:
do we want to have a default here? seems like we'd be adding the onAfterUpdate methods to an arbitrary component if one isn't provided as a parameter. I might be misunderstanding the function

Reply:
Nope, that's correct. Maybe throwing an error would be more helpful here so that it's clear that an important parameter wasn't supplied.

Reply:
yeah if the component is required for this to work we should either let the function break or throw some kind of error